plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetastoreClosure.java

```diff
@@ -217,7 +221,11 @@ public void dropColumn(String databaseName, String tableName, String columnName)
             List<String> columnNames,
             TupleDomain<String> partitionKeysFilter)
     {
-        return delegate.getPartitionNamesByFilter(databaseName, tableName, columnNames, partitionKeysFilter);
+        Table table = delegate.getTable(databaseName, tableName)
+                .orElseThrow(() -> new TrinoException(HIVE_TABLE_DROPPED_DURING_QUERY, "Table does not exists: " + tableName));
+        return getPartitionProjection(table)
```

Comment:
I would find the statement with if (partition projection table){ ... } else {...} more readable.

Reply:
Yep looks better now. Applied